include/networkit/graph/GraphTools.hpp

```diff
@@ -129,13 +129,86 @@ std::pair<node, node> size(const Graph &G) noexcept;
 double density(const Graph &G) noexcept;
 
 /**
- * Returns the volume (sum of the out-degree of all nodes) of the graph.
+ * Returns the volume of the input graph.
  *
  * @param G The input graph.
  * @return Volume of the graph.
+ * 
  */
 double volume(const Graph &G);
 
+/**
+ * Returns the volume (sum of the out-degree of all nodes) of a range
+ * of nodes.
+ *
+ * @param G The input graph.
+ * @param first Iterator for the starting node in the set.
+ * @param last Iterator for the last node in the set.
+ * @return volume of the graph.
+ */
+template <class InputIt>
+double volume(const Graph &G, InputIt first, InputIt last) {
+    double outVolume = 0.0;
+    bool countSelfLoopsTwice = !G.isDirected();
+    count nodesInGraph = 0;
+    InputIt actIt = first;
```

Comment:
Probably it is cleaner to simply declare `actIt` in the for loops below. E.g., `for (auto actIt = first; actIt != last; ++actIt)`